rpcserver.go

```diff
@@ -679,13 +679,50 @@ func (r *rpcServer) Start() error {
 
 		go func() {
 			rpcsLog.Infof("gRPC proxy started at %s", lis.Addr())
-			http.Serve(lis, mux)
+			if len(cfg.RestCORS) == 0 {
+				http.Serve(lis, mux)
+			} else {
+				http.Serve(lis, checkCors(mux))
+			}
 		}()
 	}
 
 	return nil
 }
 
+// checkCors allows Cross Origin Resoruce Sharing from configured origins.
+func checkCors(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+
+		// Skip CORS check if request did not send origin
+		if origin == "" {
+			return
```

Comment:
Should you still call `h.ServeHTTP(w, r)` for these non-browser http requests?

Reply:
Yeah, nice catch. Updated.